daemon/src/main/java/org/sonarlint/daemon/services/AbstractSonarLint.java

```diff
@@ -0,0 +1,139 @@
+import org.sonarsource.sonarlint.core.client.api.common.LogOutput;
+import org.sonarsource.sonarlint.core.client.api.common.analysis.ClientInputFile;
+import org.sonarsource.sonarlint.core.client.api.common.analysis.IssueListener;
+import org.sonarsource.sonarlint.daemon.proto.SonarlintDaemon.Issue;
+import org.sonarsource.sonarlint.daemon.proto.SonarlintDaemon.Issue.Severity;
+import org.sonarsource.sonarlint.daemon.proto.SonarlintDaemon.LogEvent;
+
+import java.nio.charset.Charset;
+import java.nio.file.Path;
+
+public abstract class AbstractSonarLint {
+  protected static class DefaultClientInputFile implements ClientInputFile {
+    private final Path path;
+    private final boolean isTest;
+    private final Charset charset;
+
+    protected DefaultClientInputFile(Path path, boolean isTest, Charset charset) {
+      this.path = path;
+      this.isTest = isTest;
+      this.charset = charset;
+    }
+
+    @Override
+    public Path getPath() {
+      return path;
+    }
+
+    @Override
+    public boolean isTest() {
+      return isTest;
+    }
+
+    @Override
+    public Charset getCharset() {
+      return charset;
+    }
+
+    @Override
+    public <G> G getClientObject() {
+      return null;
+    }
+  }
+
+  protected static class ProxyIssueListener implements IssueListener {
+    private final StreamObserver<Issue> observer;
+
+    protected ProxyIssueListener(StreamObserver<Issue> observer) {
+      this.observer = observer;
+    }
+
+    @Override
+    public void handle(org.sonarsource.sonarlint.core.client.api.common.analysis.Issue issue) {
+      Severity severity;
+
+      switch (issue.getSeverity()) {
+        case "MINOR":
+          severity = Severity.MINOR;
+          break;
+        case "BLOCKER":
+          severity = Severity.BLOCKER;
+          break;
+        case "INFO":
+          severity = Severity.INFO;
+          break;
+        case "CRITICAL":
+          severity = Severity.CRITICAL;
+          break;
+        case "MAJOR":
+        default:
+          severity = Severity.MAJOR;
+          break;
+      }
+
+      observer.onNext(Issue.newBuilder()
+        .setRuleKey(issue.getRuleKey())
+        .setRuleName(issue.getRuleName())
+        .setMessage(issue.getMessage())
+        .setSeverity(severity)
+        .setStartLine(issue.getStartLine())
+        .setStartLineOffset(issue.getStartLineOffset())
+        .setEndLine(issue.getEndLine())
+        .setEndLineOffset(issue.getEndLineOffset())
+        .build());
+    }
+  }
+
+  protected static class ProxyLogOutput implements LogOutput {
+    private StreamObserver<LogEvent> response;
+
+    protected void setObserver(StreamObserver<LogEvent> response) {
+      this.response = response;
+    }
+
+    @Override
+    public synchronized void log(String formattedMessage, Level level) {
+      if (level == Level.ERROR) {
+        System.err.println(formattedMessage);
+      } else {
+        System.out.println(formattedMessage);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Replace this usage of System.out or System.err by a logger. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS106)